[Joel Wong] iP

joelwongjy opened the pull request:
### MickeyPro
>“You control your destiny — you don't need magic to do it." - Walt Disney

MickeyPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/joelwongjy/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x]  Managing tasks
- [x]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

#
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

joelwongjy committed:
Add Greet, Echo, Exit

Implement skeletal version of Mickey that starts by greeting the user, echos commands entered by the user, and exits when the user types bye.

joelwongjy committed:
Add add and list

Add the ability to store whatever text entered by the user and display them back to the user when requested.

joelwongjy committed:
Add mark as done

Add the ability to mark tasks as done.
Add the ability to change the status back to not done.

joelwongjy committed:
Add todo, events, deadlines

Add support for tracking three types of tasks:

ToDos: tasks without any date/time attached to it
Deadlines: tasks that need to be done before a specific date/time
Events: tasks that start and end at a specific time

joelwongjy committed:
Add automated text ui testing

Use input/output redirection technique to semi-automate testing of Mickey

joelwongjy committed:
Rename duke to mickey


Rename duke to mickey

joelwongjy committed:
Handle errors

Allow Mickey to deal with errors such as incorrect inputs entered by the user.

joelwongjy committed:
Add delete

Add support for deleting tasks from the list.

joelwongjy committed:
Add save

joelwongjy committed:
Format dates to datetime objects

joelwongjy committed:
Merge pull request #1 from joelwongjy/branch-Level-7

Add save tasks

joelwongjy committed:
Merge pull request #2 from joelwongjy/branch-Level-8

Format dates to datetime objects

joelwongjy committed:
Refactor to use more OOP and package

joelwongjy committed:
Refactor packages

joelwongjy committed:
Add JUnit Tests

joelwongjy committed:
Handle datetime parsing error

joelwongjy committed:
Add JavaDoc comments

joelwongjy committed:
Merge pull request #3 from joelwongjy/branch-A-JavaDoc

Add JavaDoc comments

joelwongjy committed:
Fix styling

joelwongjy committed:
Merge pull request #4 from joelwongjy/branch-A-CodingStandard

Fix styling

joelwongjy committed:
Add find task


Add find task

joelwongjy committed:
Merge pull request #5 from joelwongjy/branch-Level-9

Add find task

joelwongjy committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

joelwongjy committed:
Add gradle

joelwongjy committed:
Merge pull request #6 from joelwongjy/branch-A-Gradle

Add gradle support

benmurphyy commented on src/main/java/mickey/command/AddCommand.java:
```diff
@@ -0,0 +1,65 @@
+package mickey.command;
+
+import mickey.app.MickeyException;
+import mickey.app.Storage;
+import mickey.app.Ui;
+import mickey.task.Deadline;
+import mickey.task.Event;
+import mickey.task.Task;
+import mickey.task.TaskList;
+import mickey.task.ToDo;
+
+import java.time.format.DateTimeParseException;
```
Should standard java library imports be above local imports? You can try adjusting your import order in Intelliji to fit the import order they want!

benmurphyy commented on src/main/java/mickey/command/AddCommand.java:
```diff
@@ -0,0 +1,65 @@
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws MickeyException {
+        Task newTask = null;
+        switch (cmd) {
+            case "todo":
+                newTask = new ToDo(getDescription());
+                ui.showNewTask(tasks.size() + 1, newTask, "todo");
+                break;
+            case "deadline":
+                try {
+                    newTask = new Deadline(getDescription(), getDate());
+                    ui.showNewTask(tasks.size() + 1, newTask, "deadline");
+                } catch (DateTimeParseException e) {
+                    System.out.println("Invalid date");
+                }
+                break;
+            case "event":
+                try {
+                    newTask = new Event(getDescription(), getDate());
+                    ui.showNewTask(tasks.size() + 1, newTask, "event");
+                } catch (DateTimeParseException e) {
+                    System.out.println("Invalid date");
+                }
+                break;
+            default:
+        }
+        if (newTask != null) {
+            tasks.add(newTask);
+        }
+    }
```
Should the indentation of this block be reduced by 4 spaces?

benmurphyy commented on src/main/java/mickey/command/Command.java:
```diff
@@ -0,0 +1,104 @@
+package mickey.command;
+
+import mickey.app.MickeyException;
+import mickey.app.Storage;
+import mickey.app.Ui;
+import mickey.task.TaskList;
+
+import java.util.Arrays;
+import java.util.Optional;
```
Should standard java library imports be above local imports?

benmurphyy commented on src/main/java/mickey/command/Command.java:
```diff
@@ -0,0 +1,104 @@
+ */
+public abstract class Command {
+    /** User input command. */
+    String cmd;
```
Should this variable be private?

benmurphyy commented on src/main/java/mickey/command/Command.java:
```diff
@@ -0,0 +1,104 @@
+    String cmd;
+
+    /** Optional String arguments. */
+    Optional<String> args;
```
Should this variable be private?

benmurphyy commented on src/main/java/mickey/command/FindCommand.java:
```diff
@@ -0,0 +1,44 @@
+package mickey.command;
+
+import mickey.app.MickeyException;
+import mickey.app.Storage;
+import mickey.app.Ui;
+import mickey.task.Task;
+import mickey.task.TaskList;
+
+import java.util.List;
```
Should standard java library imports be above local imports?

benmurphyy commented on src/main/java/mickey/task/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+    /**
+     * Task deadline date.
+     */
+    public LocalDateTime by;
```
Should this variable be private?

benmurphyy commented on src/main/java/mickey/task/Event.java:
```diff
@@ -0,0 +1,33 @@
+    /**
+     * Event date.
+     */
+    public LocalDateTime at;
```
Should this variable be private?

benmurphyy commented on src/main/java/mickey/task/Task.java:
```diff
@@ -0,0 +1,46 @@
+ */
+public abstract class Task {
+    /** Task description */
+    public String description;
```
Should this variable be private?

benmurphyy commented on src/main/java/mickey/task/Task.java:
```diff
@@ -0,0 +1,46 @@
+    public String description;
+
+    /** If task is done. */
+    public boolean isDone = false;
```
Should this variable be private?

benmurphyy commented on src/main/java/mickey/task/Task.java:
```diff
@@ -0,0 +1,46 @@
+    public String toString() {
+        return String.format("[%s] %s", getStatusIcon(), description);
+    }
+}
```
Should this file end with a newline?

benmurphyy commented on src/main/java/mickey/app/MickeyException.java:
```diff
@@ -0,0 +1,16 @@
+    public MickeyException(String msg) {
+        super(msg);
+    }
+}
```
Should this file end with a newline?

benmurphyy commented on src/main/java/mickey/app/Mickey.java:
```diff
@@ -0,0 +1,64 @@
+    public static void main(String[] args) {
+        new Mickey("src/main/data/tasks.txt").run();
+    }
+}
```
Should this file end with a newline?

benmurphyy reviewed (commented):
Overall I found your code easy to read and understand, just a few minor coding standard violations and it should be good!

joelwongjy committed:
Install JavaFX and create GUI

joelwongjy committed:
Merge pull request #7 from joelwongjy/branch-Level-10

Install JavaFX and create GUI

joelwongjy committed:
Move launcher directory and modify gradle build file

joelwongjy committed:
Remove scene font setting

joelwongjy committed:
Add assertions

Important assumptions hold at various points in code.

Assumptions should be checked to ensure code follows expected behavior.

Let's add assertions to Parser and Storage classes.

joelwongjy committed:
Improve code quality

Code does not follow coding standard guidelines.

Code is not as readable due to inconsistencies and poor structure.

Let's refactor to improve code quality and improve styling.

joelwongjy committed:
Merge pull request #8 from joelwongjy/branch-A-Assertions

Add assertions

joelwongjy committed:
Merge pull request #9 from joelwongjy/branch-A-CodeQuality

Improve code quality

joelwongjy committed:
Add sort extension

Current tasks can only be sorted by the order added.

Users cannot view tasks by a custom order.

Let's add a sort command to sort by alphabetical or deadline order.

joelwongjy committed:
Improve GUI

Current GUI does not differentiate user and app responses.
Profile pictures are incorrect.

Let's show different colors for user and app responses.
Add images for Mickey and Goofy as profile picture.

joelwongjy committed:
Set theme jekyll-theme-minimal

joelwongjy committed:
Create user guide

Add UI screenshots and write user guide.

joelwongjy committed:
Merge remote-tracking branch 'origin/master'

joelwongjy committed:
Remove console log statements

Print statements meant for debugging are present.

Let's remove them.

joelwongjy committed:
Fix tasks not stored

Tasks are not stored upon exit.

When executed from a jar file, save file is not created in the user's
directory. Instead, it is in a set filePath.

Let's modify the storage class to get user working directory to save
file.

joelwongjy committed:
Fix styling issues

joelwongjy committed:
Close application on bye command

Bye command does not close the application.

Let's modify user input function to exit the application after 2 seconds.
